proof/meta.go

```diff
@@ -232,6 +243,12 @@ func (m *MetaReveal) GetDecDisplay() (map[string]interface{}, uint32, error) {
 		return nil, 0, nil
 	}
 
+	// If the decimal display is set as the new TLV value, we can use that
+	// directly.
+	if decimalDisplay, isSet := m.DecimalDisplay.Unwrap(); isSet {
+		return nil, decimalDisplay, nil
```

Comment:
Before returning, should we do a sanity check that:
  * The value isn't already encoded in the asset meta JSON
  * That if it is then it matches this value

?

Reply:
I added a check for that in the RPC server, so for when we create the reveal. If someone uses a different implementation and allows encoding of different values in the JSON vs. the TLV field, IMO there's not much we can do about it and the TLV value should probably take precedence.